src/fontra/views/editor/visualization-layer-definitions.js

```diff
@@ -281,6 +279,138 @@ registerVisualizationLayerDefinition({
   },
 });
 
+registerVisualizationLayerDefinition({
+  identifier: "fontra.guidelines",
+  name: "Guidelines",
+  selectionMode: "editing",
+  userSwitchable: true,
+  defaultOn: false,
+  zIndex: 500,
+  screenParameters: {
+    fontSize: 10,
+    strokeWidth: 1,
+    strokeLength: 1000,
+    originMarkerRadius: 4,
+    strokeDash: 3,
+    margin: 5,
+    iconSize: 12,
+  },
+  colors: {
+    strokeColor: "#0006",
+    boxColor: "#FFFB",
+    color: "#000",
+  },
+  colorsDarkMode: {
+    strokeColor: "#FFF8",
+    boxColor: "#1118",
+    color: "#FFF",
+  },
+  draw: (context, positionedGlyph, parameters, model, controller) => {
+    context.font = `${parameters.fontSize}px fontra-ui-regular, sans-serif`;
+    context.textAlign = "center";
+
+    // Draw glyph guidelines
+    for (const guideline of positionedGlyph.glyph.guidelines) {
+      _drawGuideline(context, parameters, guideline);
+    }
+
+    // TODO: Font Guidelines
+  },
+});
+
+function _drawGuideline(context, parameters, guideline) {
+  context.save();
+  context.strokeStyle = parameters.strokeColor;
+  context.lineWidth = parameters.strokeWidth;
+  // move to the origin of the guideline
+  context.translate(guideline.x, guideline.y);
+  context.rotate((guideline.angle * Math.PI) / 180);
+  context.scale(1, -1);
+
+  // draw guideline origin marker
+  if (guideline.locked) {
+    context.save();
+    context.rotate((guideline.angle * Math.PI) / 180);
+    context.scale(1, -1);
+    _drawLockIcon(
+      context,
+      parameters,
+      -parameters.iconSize / 2,
+      parameters.iconSize / 2
+    );
+    context.restore();
+  } else {
+    strokeCircle(context, 0, 0, parameters.originMarkerRadius);
+  }
+
+  let textWidth;
+  let textHeight;
+  // draw name
+  if (guideline.name) {
+    const strLine = `${guideline.name}`;
+    textWidth = Math.max(context.measureText(strLine).width);
+    textHeight = Math.max(getTextHeight(context, strLine));
+
+    context.fillStyle = parameters.strokeColor;
+    context.fillText(
+      strLine,
+      -textWidth / 2 -
+        parameters.margin * 2 -
+        parameters.originMarkerRadius / 2 -
+        parameters.strokeWidth * 2,
+      textHeight / 2 - parameters.fontSize * 0.2 // move up -> visually centered
+    );
+  }
+
+  // draw the line
+  if (guideline.name !== undefined) {
```

Comment:
In the next bit, there are many calls to `strokeLineDashed` that are very repetitive. All args are always the same, except for the x1 and x2 argumens.

I suggest to build an array instead, with an item per line you need to draw, and each item is a two-item array: [x1, x2]. 

After collecting this info, you can do a trivial `for (const [x1, x2] of lines) { ...do strokeLineDashed()... }`

Reply:
I am totally with you. I refactored it.